includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1065,7 +1080,7 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 			if ( 'requires_action' === $status && $payment_information->is_merchant_initiated() ) {
 				// Allow 3rd-party to trigger some action if needed.
 				do_action( 'woocommerce_woocommerce_payments_payment_requires_action', $order, $intent_id, $payment_method, $customer_id, $charge_id, $currency );
-				$order->update_status( 'failed' );
+				$this->order_service->mark_payment_failed( $order, $intent_id, $status, $charge_id, '' );
```

Comment:
Extreme Nit Pick: I think we don't need to pass the `''` as the last parameter here, because it's the default value of `$message` if no parameter is passed to the function.